cloudwatch/cloudwatch.go

```diff
@@ -509,6 +510,15 @@ func (output *OutputPlugin) processRecord(e *Event) ([]byte, error) {
 		return nil, err
 	}
 
+	// append newline
+	data = append(data, []byte("\n")...)
+
+	if (len(data) + perEventBytes) > maximumBytesPerEvent {
+		logrus.Warnf("[cloudwatch %d] Found record with %d bytes, truncating to 256KB, logGroup=%s, stream=%s\n", output.PluginInstanceID, len(data)+perEventBytes, output.logGroupName, output.logStreamName)
+		data = data[:(maximumBytesPerEvent - len(TruncatedSuffix) - perEventBytes)]
+		data = append(data, []byte(TruncatedSuffix)...)
+	}
+
```

Comment:
I am not sure if we are checking the right thing. @PettitWesley

Reply:
@hossain-rayhan We are. We checked the code for the CW Agent. The new code is basically copied from it.